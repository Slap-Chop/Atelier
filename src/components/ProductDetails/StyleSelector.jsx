import React from 'react';

const StyleSelector = (props) => {
  return(
    <div style={{border: '1px solid green'}}>Style StyleSelector placeholder
  </div>
  )

}

export default StyleSelector;