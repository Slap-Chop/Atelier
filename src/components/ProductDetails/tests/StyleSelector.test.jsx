import React from 'react';
import {shallow} from 'enzyme';
import StyleSelector from '../StyleSelector.jsx';
import toJson from 'enzyme-to-json';

const testProduct = {
  category: "Jackets",
  default_price: "140.00",
  description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
  id: 40344,
  name: "Camo Onesie",
  slogan: "Blend in to your crowd"
}

const testStyles = [{
  'default?': true,
  name: "Forest Green & Black",
  original_price: "140.00",
  photos: [{
    thumbnail_url: "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
    url: "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
  }],
  sale_price: null,
  skus: {1394769: {quantity: 8, size: 'XS'}},
  style_id: 240500
},
{
  name: "Forest Green & Black",
  original_price: "140.00",
  photos: [{
    thumbnail_url: "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
    url: "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
  }],
  sale_price: null,
  skus: {1394769: {quantity: 8, size: 'XS'}},
  style_id: 240500
}]

const mockClick = jest.fn();

const testSelector = shallow(<StyleSelector
  currentProduct={testProduct}
  currentStyle={testStyles[0]}
  click={mockClick}
  id={240500}
  styles={testStyles}/>)

  test('Style Selector should render properly and stay consistent', () => {
    expect(toJson(testSelector)).toMatchSnapshot();
  })