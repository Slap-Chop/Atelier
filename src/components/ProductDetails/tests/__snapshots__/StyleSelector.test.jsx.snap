// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Style Selector should render properly and stay consistent 1`] = `
<div
  className="productBorder"
>
  <div
    className="flex"
  >
    <p>
      $
      140.00
    </p>
    <p
      className="salePrice"
    />
  </div>
  <div>
    Style &gt; 
    Forest Green & Black
  </div>
  <div
    className="flex"
    key="0"
  >
    <StyleComponent
      click={[MockFunction]}
      current="65px"
      key="0"
      style={
        Object {
          "default?": true,
          "name": "Forest Green & Black",
          "original_price": "140.00",
          "photos": Array [
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
              "url": "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80",
            },
          ],
          "sale_price": null,
          "skus": Object {
            "1394769": Object {
              "quantity": 8,
              "size": "XS",
            },
          },
          "style_id": 240500,
        }
      }
      weight={600}
    />
    <StyleComponent
      click={[MockFunction]}
      key="1"
      style={
        Object {
          "name": "Forest Green & Black",
          "original_price": "140.00",
          "photos": Array [
            Object {
              "thumbnail_url": "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
              "url": "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80",
            },
          ],
          "sale_price": null,
          "skus": Object {
            "1394769": Object {
              "quantity": 8,
              "size": "XS",
            },
          },
          "style_id": 240500,
        }
      }
    />
  </div>
</div>
`;
